src/sentry/snuba/discover.py

```diff
@@ -735,6 +735,128 @@ def get_facets(query, params, limit=10, referrer=None):
+    params (Dict[str, str]) Filtering parameters with start, end, project_id, environment
+    limit (int) The number of records to fetch.
+    referrer (str|None) A referrer string to help locate the origin of this query.
+
+    Returns Sequence[FacetResult]
+    """
+    with sentry_sdk.start_span(
+        op="discover.discover", description="facets.filter_transform"
+    ) as span:
+        span.set_data("query", query)
+        snuba_filter = get_filter(query, params)
+
+        # Resolve the public aliases into the discover dataset names.
+        snuba_filter, translated_columns = resolve_discover_aliases(snuba_filter)
+
+    # Exclude tracing tags as they are noisy and generally not helpful.
+    # TODO(markus): Tracing tags are no longer written but may still reside in DB.
+    excluded_tags = ["tags_key", "NOT IN", ["trace", "trace.ctx", "trace.span", "project"]]
+
+    # Sampling keys for multi-project results as we don't need accuracy
+    # with that much data.
+    sample = len(snuba_filter.filter_keys["project_id"]) > 2
+
+    with sentry_sdk.start_span(op="discover.discover", description="facets.frequent_tags"):
+        # Get the most frequent tag keys
+        key_names = raw_query(
+            aggregations=[["stddevSamp", aggregate_column, "stddev"]],
+            start=snuba_filter.start,
+            end=snuba_filter.end,
+            conditions=snuba_filter.conditions,
+            filter_keys=snuba_filter.filter_keys,
+            orderby=["-stddev", "tags_key"],
+            groupby="tags_key",
+            having=[excluded_tags],
+            dataset=Dataset.Discover,
+            limit=limit,
+            referrer=referrer,
+            turbo=sample,
+        )
+        top_tags = [r["tags_key"] for r in key_names["data"]]
+        if not top_tags:
+            return []
+
+    results = []
+
+    sampling_enabled = True
+    sample_rate = 0.1 if sampling_enabled else None
+
+    max_aggregate_tags = 20
+    individual_tags = []
+    aggregate_tags = []
+    for i, tag in enumerate(top_tags):
+        if tag == "environment":
+            # Add here tags that you want to be individual
+            individual_tags.append(tag)
+        elif i >= len(top_tags) - max_aggregate_tags:
+            aggregate_tags.append(tag)
+        else:
+            individual_tags.append(tag)
+
+    with sentry_sdk.start_span(
+        op="discover.discover", description="facets.individual_tags"
+    ) as span:
+        span.set_data("tag_count", len(individual_tags))
+        for tag_name in individual_tags:
+            tag = f"tags[{tag_name}]"
+            tag_values = raw_query(
+                aggregations=[[aggregate_function, aggregate_column, "aggregate"]],
+                conditions=snuba_filter.conditions,
+                start=snuba_filter.start,
+                end=snuba_filter.end,
+                filter_keys=snuba_filter.filter_keys,
+                orderby=[],
```

Comment:
Actually I intentionally left this blank here because a.) largest aggregate isn't necessarily the best, since I want large and small aggregates to show the largest diff and  b.) didn't want to do any more ordering in case there is a perf impact. I just left it as unordered then. In the future we can figure out a smarter way of getting the most useful tag values out.

Reply:
Not sure I understand the intention of excluding an orderby, we'll likely get inconsistent results in prod, could we sort by count instead so that we're looking at the relatively common values at least?